### PR TITLE
[proof-new] Add test cases to exercise bitvector proof reconstruction rules.

### PR DESCRIPTION
**Pull request by vleni**
1. Remove `bvurem` and `bvudiv` RARE rules that handle negative cases - they are never invoked. More specifically, the corresponding rules `UremPow2` and `UdivPow2` require the divisor to be non-negative. The negative cases are removed, and the cases for dividing by 1 are handled by `bv-{udiv,urem}-one`.
2. Add some test cases that are supposed to be filled by exisitng rules, but either a roundabout method is used or the hole doesn't get filled.
3. Fixed a bug in `bv-usubo-eliminate`
4. Fixed a bug in `bv-extract-mult-leading-bit`

There remain 3 cases of bv reconstruction rules unable to fill holes generated by the rewriter.

### CODE REVIEW

File: src/theory/bv/rewrites [hunk 1]
```diff
@@ -591,39 +591,24 @@
...
 
 ; (a udiv 2^k) ==> 0_k a[n-1: k]
-(define-cond-rule bv-udiv-pow2-1p
+(define-cond-rule bv-udiv-pow2-not-one
   ((x ?BitVec) (v Int) (n Int))
   (def (power (int.log2 v)))
   (and (int.ispow2 v) (> v 1))
   (bvudiv x (bv v n))
   (concat (bv 0 power) (extract (- n 1) power x)))
-(define-cond-rule bv-udiv-pow2-1n
```
Comment (Lachnitt) [inline src/theory/bv/rewrites:610]: Why are these redundant? The only other rule on bvudiv seems to be bv-udiv-pow2-1p?

**Review by Lachnitt — COMMENTED** [review]
I have one question otherwise it looks good to me.

File: src/theory/bv/rewrites (hunk 1 above)
Comment (vleni) [reply src/theory/bv/rewrites:610]: If you look at `UdivPow2`, it never applies for negative cases.

Comment (vleni) [reply src/theory/bv/rewrites:610]: I added an explanation in the PR

**Review by ajreynol — APPROVED** [review]

**Pull request merged**